examples/spouts/spout-marker/README.md

```diff
@@ -0,0 +1,328 @@
+
+* Pachyderm 1.9.10 or later
+* Terminal
+
+## Pipeline Overview
+
+In this example, we will use a simple spout pipeline that
+will add dots into a spout marker file. Here is how the
+spout file will look like:
+
+```bash
+.
+..
+...
+....
+.....
+......
+.......
+```
+
+The script runs every thirty seconds and appends a dot (`.`) and a new line
+to the marker file creating a half pyramid pattern.
+
+After running the pipeline for some time, we will modify the Python
+script so that it adds the star (`*`) symbol instead of a dot. The
+resulting file should look like this:
+
+```bash
+
+.
+..
+...
+....
+.....
+......
+.......*
+.......**
+.......***
+.......****
+.......*****
+```
+
+## Step 1: Build the Docker Image
+
+Pachyderm uses Docker images that you specify in your
+pipeline to create Kubernetes pods that run your code.
+For this example, we will use a very simple [Dockerfile](./Dockerfile)
+that pulls a basic Python image and adds your code to
+the container that will run your code.
+
+To build a Docker image, complete the following steps:
+
+1. Clone this repository:
+
+   ```bash
+   git clone git@github.com:pachyderm/pachyderm.git
+   ```
+
+1. Change the directory to `examples/spouts/spout-marker/`.
+
+1. Build and a tag a Docker image from the Dockerfile in this directory:
+
+   ```bash
+   docker build --tag spout-marker:v1 .
+   ```
+
+   **Note:** Do not forget the dot in the end!
+
+1. Push the Docker image to an image registry.
+
+   * If you are using `minikube`, for testing you can just
+   transfer your local image to a `minikube` VM:
+
+     ```bash
+     docker save spout-marker:v1 | (\
+     eval $(minikube docker-env)
+     docker load
+     )
+     ```
+
+1. Proceed to [Step 2](#step-2-create-the-pipeline).
+
+## Step 2: Create the Pipeline
+
+Because spouts do not have an input and consume data from an
+outside source, you do not need to create a Pachyderm
+repository for this example. For this example, we do not need
+to set up any messaging system, because the Python script will
+generate it for us.
+
+However, you still need to create
+the spout pipeline with a marker file.
+The pipeline specification for this example is stored in
+[spout-marker-pipeline.json](./spout-marker-pipeline.json).
+
+The Python script that we will use for this example is stored in
+[spout-marker-example.py](./spout-marker-example.py).
+
+When you create a spout pipeline with a marker file, Pachyderm
+creates a separate branch for the spout marker and stores the
+marker file in that branch.
+
+To create a spout pipeline, complete the following steps:
+
+1. Create a spout pipeline:
+
+   ```bash
+   # spout-marker-pipeline.json
+   {
+   "pipeline": {
+    "name": "spoutmarker"
+   },
+   "transform": {
+     "cmd": [ "python3", "/spout-marker-example.py" ],
+     "image": "spout-marker:v1"
+   },
+   "spout": {
+     "marker": "mymark"
+    }
+   }
+   ```
+
+   **Note:** In the `spout` section, you have a key-value pair
```

Comment:
Maybe worth making it more explicit that `mymark` here will be the name of your marker file (or more accurately, will be make all files beginning with `mymark` into marker files, so you could have `mymark01` and `mymark02` if you needed two files)